app/assets/javascripts/static.coffee

```diff
@@ -9,3 +9,14 @@ $(".static.index").ready ->
 $(".static.quiz1_verify_details").ready ->
   $("#primary_products").tagit
     allowSpaces: true
+
+$(".static.match_detail").ready ->
+  $(".contact_button").on "click", (e) ->
+    $(".hidden_contact_info_container").toggleClass("hidden_overlay")
+    e.stopPropagation()
+
+  $(".contact_info_card").on "click", (e) ->
+    e.stopPropagation()
+
+  $("body").on "click", (e) ->
+    $(".hidden_contact_info_container").addClass("hidden_overlay")
```

Comment:
before adding the class, it would be good to check `hasClass` so you don't add a bunch of repetitive classes to it on every body click